File: src/main/kotlin/com/pleon/chopchop/ThemeUtil.java
package com.pleon.chopchop;

import javafx.beans.property.Property;
import javafx.beans.property.SimpleObjectProperty;
import javafx.scene.Node;

public class ThemeUtil {

    public enum Theme {
        DARK, LIGHT;
    }

    private static Property<Theme> currentTheme = new SimpleObjectProperty<>(Theme.DARK);

    public static Property<Theme> getThemeProperty() {
        return currentTheme;
    }

    public static void toggleTheme() {
        if (currentTheme.getValue() == Theme.DARK) {
            currentTheme.setValue(Theme.LIGHT);
        } else {
            currentTheme.setValue(Theme.DARK);
        }
    }

    public static void applyTheme(Node root) {
        if (currentTheme.getValue() == Theme.LIGHT) {
            root.getStyleClass().removeAll("dark");
        } else {
            root.getStyleClass().add("dark");
        }
    }
}

File: src/main/kotlin/com/pleon/chopchop/ThemeUtil.kt
package com.pleon.chopchop

import com.pleon.chopchop.ThemeUtil.Theme.DARK
import com.pleon.chopchop.ThemeUtil.Theme.LIGHT
import javafx.beans.InvalidationListener
import javafx.beans.property.SimpleObjectProperty
import javafx.scene.Node

object ThemeUtil {

    private enum class Theme {
        DARK, LIGHT
    }

    private var theme = SimpleObjectProperty(DARK)

    fun onThemeChanged(listener: InvalidationListener) {
        theme.addListener(listener)
    }

    fun toggleTheme() {
        theme.value = if (theme.value == DARK) LIGHT else DARK
    }

    fun applyTheme(node: Node) {
        if (theme.value == LIGHT) {
            node.styleClass.removeAll("dark")
        } else {
            node.styleClass.add("dark")
        }
    }
}
